proxy.go

```diff
@@ -58,29 +57,20 @@ func (hpd *httpProxyDialer) Dial(network string, addr string) (net.Conn, error)
 	}
 
 	if err := connectReq.Write(conn); err != nil {
-		if err := conn.Close(); err != nil {
-			log.Printf("httpProxyDialer: failed to close connection: %v", err)
-		}
-		return nil, err
+		return nil, errors.Join(conn.Close(), err)
```

Comment:
Reorder so that primary error is first.

			return nil, errors.Join(err, conn.Close())